mutiny-core/src/fedimint.rs

```diff
@@ -0,0 +1,271 @@
+use fedimint_ln_client::{
+    InternalPayState, LightningClientInit, LightningClientModule, LnPayState, LnReceiveState,
+};
+use fedimint_mint_client::MintClientInit;
+use fedimint_wallet_client::WalletClientInit;
+use futures_util::StreamExt;
+use lightning::util::logger::Logger;
+use lightning::{log_debug, log_info};
+use lightning_invoice::Bolt11Invoice;
+use serde::{Deserialize, Serialize};
+use std::{
+    collections::HashMap,
+    str::FromStr,
+    sync::{atomic::AtomicBool, Arc, RwLock},
+};
+
+const FEDIMINT_CLIENT_NONCE: &[u8] = b"Fedimint Client Salt";
+
+impl From<LnReceiveState> for HTLCStatus {
+    fn from(state: LnReceiveState) -> Self {
+        match state {
+            LnReceiveState::Created => HTLCStatus::Pending,
+            LnReceiveState::Claimed => HTLCStatus::Succeeded,
+            LnReceiveState::WaitingForPayment { .. } => HTLCStatus::Pending,
+            LnReceiveState::Canceled { .. } => HTLCStatus::Failed,
+            LnReceiveState::Funded => HTLCStatus::InFlight,
+            LnReceiveState::AwaitingFunds => HTLCStatus::InFlight,
+        }
+    }
+}
+
+impl From<InternalPayState> for HTLCStatus {
+    fn from(state: InternalPayState) -> Self {
+        match state {
+            InternalPayState::Funding => HTLCStatus::InFlight,
+            InternalPayState::Preimage(_) => HTLCStatus::Succeeded,
+            InternalPayState::RefundSuccess { .. } => HTLCStatus::Failed,
+            InternalPayState::RefundError { .. } => HTLCStatus::Failed,
+            InternalPayState::FundingFailed { .. } => HTLCStatus::Failed,
+            InternalPayState::UnexpectedError(_) => HTLCStatus::Failed,
+        }
+    }
+}
+
+impl From<LnPayState> for HTLCStatus {
+    fn from(state: LnPayState) -> Self {
+        match state {
+            LnPayState::Created => HTLCStatus::Pending,
+            LnPayState::Canceled => HTLCStatus::Failed,
+            LnPayState::Funded => HTLCStatus::InFlight,
+            LnPayState::WaitingForRefund { .. } => HTLCStatus::InFlight,
+            LnPayState::AwaitingChange => HTLCStatus::InFlight,
+            LnPayState::Success { .. } => HTLCStatus::Succeeded,
+            LnPayState::Refunded { .. } => HTLCStatus::Failed,
+            LnPayState::UnexpectedError { .. } => HTLCStatus::Failed,
+        }
+    }
+}
+
+// This is the FedimintStorage object saved to the DB
+#[derive(Debug, Serialize, Deserialize, Clone, Default)]
+pub struct FedimintStorage {
+    pub fedimints: HashMap<String, FedimintIndex>,
+    #[serde(default)]
+    pub version: u32,
+}
+
+// This is the FedimintIdentity that refer to a specific node
+// Used for public facing identification.
+pub struct FedimintIdentity {
+    pub uuid: String,
+    pub federation_id: FederationId,
+}
+
+// This is the FedimintIndex reference that is saved to the DB
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone, Default)]
+pub struct FedimintIndex {
+    pub child_index: u32,
+    pub federation_code: String,
```

Comment:
You should be able to make this `InviteCode` so you don't need to parse it later

Reply:
looks like there is a bunch of places where you are using `String` would be nice to use `InviteCode` instead